File: src/shared/schemas.ts
import { z } from 'zod';

export const CoreEntitySchema = z.object({
  id: z.string(),
  type: z.string(),
  status: z.enum(['active', 'inactive', 'archived']),
  attributes: z.record(z.unknown()),
  created_at: z.string().datetime(),
  updated_at: z.string().datetime(),
});

export const ConfigDefinitionSchema = z.object({
  config_id: z.string(),
  namespace: z.string(),
  version: z.number().int().min(1),
  parameters: z.record(z.unknown()),
  enabled: z.boolean(),
  applied_at: z.string().datetime(),
  rollback_from: z.number().int().min(1).optional(),
});

export const RunInstanceSchema = z.object({
  run_id: z.string(),
  entity_id: z.string(),
  phase: z.enum(['pending', 'executing', 'completed', 'failed', 'rollback']),
  progress: z.number().min(0).max(1),
  started_at: z.string().datetime(),
  completed_at: z.string().datetime().nullable(),
  error_detail: z.string().nullable(),
  metadata: z.record(z.unknown()),
});

export const StatsSnapshotSchema = z.object({
  snapshot_id: z.string(),
  timestamp: z.string().datetime(),
  metrics: z.object({
    throughput: z.number(),
    latency_p99: z.number(),
    error_rate: z.number(),
  }).catchall(z.number()),
  dimensions: z.object({
    host: z.string(),
    region: z.string(),
  }).catchall(z.string()),
});

export const TaskCreateSchema = z.object({
  type: z.string(),
  config: z.record(z.unknown()),
  labels: z.record(z.string()),
});

export const BatchOperationSchema = z.object({
  operations: z.array(
    z.object({
      action: z.enum(['start', 'stop', 'restart', 'delete']),
      id: z.string(),
    })
  ),
});

export const AuthTokenSchema = z.object({
  user_id: z.string(),
  roles: z.array(z.string()),
  permissions: z.array(z.string()),
  tenant_id: z.string(),
  exp: z.number(),
});

export const LogLevelSchema = z.enum(['debug', 'info', 'warn', 'error', 'fatal']);

export type CoreEntity = z.infer<typeof CoreEntitySchema>;
export type ConfigDefinition = z.infer<typeof ConfigDefinitionSchema>;
export type RunInstance = z.infer<typeof RunInstanceSchema>;
export type StatsSnapshot = z.infer<typeof StatsSnapshotSchema>;
export type TaskCreate = z.infer<typeof TaskCreateSchema>;
export type BatchOperation = z.infer<typeof BatchOperationSchema>;
export type AuthToken = z.infer<typeof AuthTokenSchema>;
export type LogLevel = z.infer<typeof LogLevelSchema>;

File: app/models/schemas.py
from datetime import datetime
from enum import Enum
from typing import Any, Dict, List, Optional
from uuid import uuid4

from pydantic import BaseModel, Field


class ResourceStatus(str, Enum):
    PROVISIONING = "provisioning"
    RUNNING = "running"
    STOPPED = "stopped"
    FAILED = "failed"
    COMPLETED = "completed"


class ResourceCreate(BaseModel):
    type: str = Field(..., description="资源类型，如 workflow, database, storage 等")
    config: Dict[str, Any] = Field(default_factory=dict, description="资源配置参数")
    labels: Dict[str, str] = Field(default_factory=dict, description="资源标签")


class ResourceResponse(BaseModel):
    code: int = 201
    data: Dict[str, Any]


class BatchOperation(BaseModel):
    action: str
    id: str


class BatchRequest(BaseModel):
    operations: List[BatchOperation]


class BatchResultItem(BaseModel):
    id: str
    success: bool
    message: Optional[str] = None


class BatchResult(BaseModel):
    code: int = 200
    data: Dict[str, Any]


class EntityModel(BaseModel):
    id: str = Field(default_factory=lambda: f"ent_{uuid4().hex[:8]}")
    type: str
    status: str = "completed"
    attributes: Dict[str, Any] = Field(default_factory=dict)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)


class ConfigModel(BaseModel):
    config_id: str = Field(default_factory=lambda: f"cfg_{uuid4().hex[:8]}")
    namespace: str = "default"
    version: int = 1
    parameters: Dict[str, Any] = Field(default_factory=dict)
    enabled: bool = True
    applied_at: Optional[datetime] = None


class RunPhase(str, Enum):
    INITIALIZING = "initializing"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"


class RunModel(BaseModel):
    run_id: str = Field(default_factory=lambda: f"run_{uuid4().hex[:8]}")
    entity_id: str
    phase: RunPhase = RunPhase.INITIALIZING
    progress: float = 0.0
    started_at: datetime = Field(default_factory=datetime.utcnow)
    completed_at: Optional[datetime] = None
    error_detail: Optional[str] = None


class MetricsSnapshotModel(BaseModel):
    snapshot_id: str = Field(default_factory=lambda: f"snap_{uuid4().hex[:8]}")
    timestamp: datetime = Field(default_factory=datetime.utcnow)
    metrics: Dict[str, float] = Field(default_factory=dict)
    dimensions: Dict[str, str] = Field(default_factory=dict)


class ErrorResponse(BaseModel):
    code: int
    error: str
    details: Optional[Dict[str, Any]] = None
